src/plugins/chart_expressions/expression_legacy_metric/public/components/with_auto_scale.tsx

```diff
@@ -120,6 +120,16 @@ export function withAutoScale<T>(WrappedComponent: ComponentType<T>) {
           ref={childrenRef}
           style={{
             transform: `scale(${scale || 0})`,
+            ...(parentDimensions.width &&
+            scale &&
+            autoScaleParams?.autoScaleMetricAlignment &&
+            autoScaleParams?.autoScaleMetricAlignment !== 'center'
+              ? {
+                  position: 'relative',
+                  [autoScaleParams.autoScaleMetricAlignment]:
+                    (1 - scale) * parentDimensions.width * scale * -1,
```

Comment:
Could you add a comment explaining the goal of this equation? It is hard to understand at first sight.